### PR TITLE
Add which-key major mode menu descriptions.

### PR DESCRIPTION
**Pull request by jmrah**
#17

## Description
Please explain the changes you made here.

## Checklist
- [x] I have read
[CONTRIBUTING](https://github.com/MarcoIeni/intellimacs/blob/master/docs/CONTRIBUTING.org)
guidelines.

#### [CHANGELOG](https://github.com/MarcoIeni/intellimacs/blob/master/docs/CHANGELOG.org):
- [ ] Updated
- [ ] I will update it after this PR has been discussed
- [x] No need to update

#### [KEYBINDINGS](https://github.com/MarcoIeni/intellimacs/blob/master/docs/KEYBINDINGS.org):
- [ ] Updated
- [ ] I will update it after this PR has been discussed
- [x] No need to update

#### [PLUGINS](https://github.com/MarcoIeni/intellimacs/blob/master/docs/PLUGINS.org):
- [ ] Updated
- [ ] I will update it after this PR has been discussed
- [x] No need to update

### CODE REVIEW

File: major/leader.vim [hunk 1]
```diff
@@ -1,5 +1,5 @@
 " Reformat whole buffer
-nnoremap <leader>m=    :action ReformatCode<CR>
-
+let g:WhichKeyDesc_Major_Format_ReformatSelection = "<leader>m== reformat-selection"
+nnoremap <leader>m==    :action ReformatCode<CR>
```
Comment (jmrah) [inline major/leader.vim:3]: Is this shortcut correct?

File: major/debug.vim [hunk 2]
```diff
@@ -1,39 +1,49 @@
 " View Breakpoints
+let g:WhichKeyDesc_Major_Debug_ViewBreakkpoints = "<leader>mdB view-breakpoints"
```
Comment (marcoieni) [inline major/debug.vim]:
```suggestion
let g:WhichKeyDesc_Major_Debug_ViewBreakpoints = "<leader>mdB view-breakpoints"
```

File: major/projects.vim [hunk 3]
```diff
@@ -1,19 +1,24 @@
 " Make project (build)
+let g:WhichKeyDesc_Major_Package_Compile = "<leader>mpb compile-project"
```
Comment (marcoieni) [inline major/projects.vim]: why do you call them `Package` instead of `Project`?

Comment (jmrah) [reply major/projects.vim]: Hmm...I can't remember why I choose package.  I think maybe I saw SPC m p as "+package" in my spacemacs editor, but....projects is a better fit here, and is more consistent with how everything else is named.

I will change it over to Project.

**Pull request merged**